src/ImageSharp/Formats/Qoi/QoiDecoderCore.cs

```diff
@@ -0,0 +1,287 @@
+                        {
+                            // Getting one pixel from previously seen pixels
+                            case QoiChunk.QoiOpIndex:
+                                readPixel = previouslySeenPixels[operationByte];
+                                pixel.FromRgba32(readPixel);
+                                break;
+
+                            // Get one pixel from the difference (-2..1) of the previous pixel
+                            case QoiChunk.QoiOpDiff:
+                                byte redDifference = (byte)((operationByte & 0b00110000) >> 4),
+                                    greenDifference = (byte)((operationByte & 0b00001100) >> 2),
+                                    blueDifference = (byte)(operationByte & 0b00000011);
+                                readPixel = previousPixel with
+                                {
+                                    R = (byte)((previousPixel.R + (redDifference - 2)) % 256),
+                                    G = (byte)((previousPixel.G + (greenDifference - 2)) % 256),
+                                    B = (byte)((previousPixel.B + (blueDifference - 2)) % 256)
+                                };
+                                pixel.FromRgba32(readPixel);
+                                pixelArrayPosition = GetArrayPosition(readPixel);
+                                previouslySeenPixels[pixelArrayPosition] = readPixel;
+                                break;
+
+                            // Get green difference in 6 bits and red and blue differences
+                            // depending on the green one
+                            case QoiChunk.QoiOpLuma:
+                                byte diffGreen = (byte)(operationByte & 0b00111111),
+                                    currentGreen = (byte)((previousPixel.G + (diffGreen - 32)) % 256),
+                                    nextByte = (byte)stream.ReadByte(),
+                                    diffRedDG = (byte)(nextByte >> 4),
+                                    diffBlueDG = (byte)(nextByte & 0b00001111),
+                                    currentRed = (byte)((diffRedDG - 8 + (diffGreen - 32) + previousPixel.R) % 256),
+                                    currentBlue = (byte)((diffBlueDG - 8 + (diffGreen - 32) + previousPixel.B) % 256);
+                                readPixel = previousPixel with { R = currentRed, B = currentBlue, G = currentGreen };
+                                pixel.FromRgba32(readPixel);
+                                pixelArrayPosition = GetArrayPosition(readPixel);
+                                previouslySeenPixels[pixelArrayPosition] = readPixel;
+                                break;
+
+                            // Repeating the previous pixel 1..63 times
+                            case QoiChunk.QoiOpRun:
+                                byte repetitions = (byte)(operationByte & 0b00111111);
```

Comment:
It's better to leave this be an `int`:

```suggestion
                                int repetitions = operationByte & 0b00111111;
```